File: base.preferences/src/main/java/alvin/base/preferences/main/views/MainActivity.java
package alvin.base.preferences.main.views;

import android.content.Intent;
import android.os.Bundle;
import android.widget.Button;

import alvin.base.preferences.R;
import alvin.base.preferences.original.views.OriginalActivity;
import alvin.base.preferences.storebox.views.StoreBoxActivity;
import androidx.appcompat.app.AppCompatActivity;
import butterknife.ButterKnife;
import butterknife.OnClick;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        ButterKnife.bind(this);
    }

    @OnClick(R.id.btn_preferences_original)
    public void onOriginalClick(Button b) {
        startActivity(new Intent(this, OriginalActivity.class));
    }

    @OnClick(R.id.btn_preferences_storebox)
    public void onStoreBoxClick(Button b) {
        startActivity(new Intent(this, StoreBoxActivity.class));
    }
}

File: ui.listing/src/main/java/alvin/ui/listing/MainActivity.kt
package alvin.ui.listing

import alvin.ui.listing.list.views.ListMainActivity
import android.content.Intent
import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import android.widget.Button
import butterknife.ButterKnife
import butterknife.OnClick

class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        ButterKnife.bind(this)
    }

    @OnClick(R.id.btn_list, R.id.btn_grid)
    fun onButtonClick(b: Button) {
        val intent: Intent? = when (b.id) {
            R.id.btn_list -> Intent(this, ListMainActivity::class.java)
            else -> null
        }

        if (intent != null) {
            startActivity(intent)
        }
    }
}
